### PR TITLE
fix: delete git empty section if exists

### PR DESCRIPTION
**Pull request by ShawkyZ**
### Description
* If loading git Config fails because of an empty subsection error, attempt to delete the subsection.
* Before persisting a subsection ensure that the values are not empty.

### Checklist

- [ ] Tests added and all succeed
- [ ] Linted
- [ ] README.md updated, if user-facing
- [ ] License file updated, if new 3rd-party dependency is introduced

### CODE REVIEW

File: internal/git_config/git_config.go [hunk 1]
```diff
@@ -117,12 +129,18 @@ func getLocalBranches(repository *git.Repository) ([]string, error) {
 }
 
 func SetBaseBranch(logger *zerolog.Logger, config []types.FolderConfig) {
+	time.Sleep(5 * time.Second)
```
Comment (ShawkyZ) [inline internal/git_config/git_config.go:132]: delete time.sleep before merging

File: internal/git_config/git_config.go [hunk 2]
```diff
@@ -134,3 +152,50 @@ func SetOption(logger *zerolog.Logger, folderPath, key string, value string) {
 		logger.Error().Err(err).Msgf("could not store %s=%s configuration for folder %s", key, value, folderPath)
 	}
 }
+
+func GitRepoFolderPath(folderPath string) (string, error) {
```
Comment (ShawkyZ) [inline internal/git_config/git_config.go:156]: moved this here because of circular dependency

File: internal/git_config/git_config.go [hunk 3]
```diff
@@ -134,3 +152,50 @@ func SetOption(logger *zerolog.Logger, folderPath, key string, value string) {
...
+	if !strings.HasSuffix(repoPath, ".git") {
+		repoPath = filepath.Join(repoPath, ".git")
+	}
+
+	return repoPath, nil
+}
+
+// DeleteEmptySnykSubsection This is a migration function to be executed if empty subsections exists
+func DeleteEmptySnykSubsection() error {
+	gitFolderPath, err := GitRepoFolderPath("")
```
Comment (ShawkyZ) [inline internal/git_config/git_config.go:180]: go-git treats empty folder as the current folder. which is annoying.

here I'm loading folderPath="" to reproduce the same behavior that we might get.

File: internal/git_config/git_config.go (hunk 2 above)
Comment (bastiandoetsch) [reply internal/git_config/git_config.go:156]: The name could be a bit more expressive... Maybe `GitRepoPath`?

File: internal/git_config/git_config.go [hunk 4]
```diff
@@ -134,3 +152,50 @@ func SetOption(logger *zerolog.Logger, folderPath, key string, value string) {
...
+	if err != nil {
+		return err
+	}
+	configPath := filepath.Join(gitFolderPath, "config")
+	cfg, err := ini.Load(configPath)
+	if err != nil {
+		return err
+	}
+	// Construct the section name with the empty subsection
+	sectionToDelete := fmt.Sprintf("%s \"\"", mainSection)
```
Comment (bastiandoetsch) [inline internal/git_config/git_config.go:190]: using backticks might help readibility here

File: internal/vcs/git_utils.go
Comment (bastiandoetsch) [inline internal/vcs/git_utils.go:1]: Maybe we should just merge git-config package with vcs?

**Review by bastiandoetsch — APPROVED** [review]

**Pull request merged**